### PR TITLE
[PPML] Apply ehsm in occlum1

### PR DESCRIPTION
**Pull request by hzjane**
## Description
Apply EHSM attestation in occlum.

### 1. Why the change?
To Apply EHSM attestation in occlum

### 2. User API changes
Using attestation will need more parameters。

### 3. Summary of the change 
Add entire ehsm process like gramine.

### 4. How to test?
- [x] Application test

### CODE REVIEW

File: ppml/trusted-big-data-ml/scala/docker-occlum/get_quote_on_ppml.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -e
+
+BLUE='\033[1;34m'
+NC='\033[0m'
+
+if [[ $1 == "musl" ]]; then
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/get_quote_on_ppml.sh:7]: Remove musl build TO reduce option

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh [hunk 2]
```diff
@@ -102,13 +102,24 @@ init_instance() {
            echo "[ERROR] Attestation set to true but NO PCCS"
            exit 1
         else
-           export ENABLE_SGX_DEBUG=false
-           sed -i "s#https://localhost:8081/sgx/certification/v3/#${PCCS_URL}#g" /etc/sgx_default_qcnl.conf
+           echo 'PCCS_URL='${PCCS_URL}'/sgx/certification/v3/' > /etc/sgx_default_qcnl.conf
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:105]: Only do this when ATTESTATION=true. https://github.com/intel-analytics/BigDL/blob/main/ppml/trusted-big-data-ml/python/docker-graphene/entrypoint.sh#L79

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh [hunk 3]
```diff
@@ -386,6 +437,19 @@ id=$([ -f "$pid" ] && echo $(wc -l < "$pid") || echo "0")
 arg=$1
 case "$arg" in
     init)
+       export RUNTIME_ENV="native"
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:440]: Why do we need to separate driver and executor init?

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh (hunk 2 above)
Comment (hzjane) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:105]: it just works as you say.

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh (hunk 3 above)
Comment (hzjane) [reply ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:440]: for next steps, now each executor have to register again.

File: ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh [hunk 4]
```diff
@@ -165,31 +176,71 @@ build_spark() {
...
-       occlum build --image-key /opt/occlum_spark/data/image_key
-       build_initfs
-    else
-       occlum build
+        if [[ $PCCS_URL == "" ]]; then
+            echo "[ERROR] Attestation set to true but NO PCCS"
+            exit 1
+        else
+            if [[ $RUNTIME_ENV == "driver" || $RUNTIME_ENV == "native" ]]; then
+                #verify ehsm service
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/run_spark_on_occlum_glibc.sh:188]: verify ehsm service is a bi-direction attestation. We need another env to enable this.

File: ppml/trusted-big-data-ml/scala/docker-occlum/verify-attestation-service.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    exit 1
+fi
+if [[ -z "$API_KEY" ]]; then
+    echo "[ERROR] API_KEY is not set!"
+    echo "[INFO] PPML Application Exit!"
+    exit 1
+fi
+if [[ -z "$CHALLENGE" ]]; then
+    echo "[ERROR] CHALLENGE is not set!"
+    echo "[INFO] PPML Application Exit!"
```
Comment (qiyuangong) [inline ppml/trusted-big-data-ml/scala/docker-occlum/verify-attestation-service.sh:26]: Using default challenge if user doesn't set.

**Review by qiyuangong — APPROVED** [review]
LGTM

**Pull request merged**